scalding-core/src/main/scala/com/twitter/scalding/FileSource.scala

```diff
@@ -142,6 +142,31 @@ object FileSource {
     !glob(globPath, conf, SuccessFileFilter).isEmpty
   }
 
+  def allPathsCommitted(globPath: String, conf: Configuration): Boolean = {
+    case class PathUsage(val path: Path, committed: Boolean)
+
+    val usedDirs: Iterable[PathUsage] = glob(globPath, conf, AcceptAllPathFilter)
+      .map { fileStatus: FileStatus =>
+        val dir =
+          if (fileStatus.isDirectory)
+            fileStatus.getPath
+          else
+            fileStatus.getPath.getParent
+
+        if (fileStatus.isFile && SuccessFileFilter.accept(fileStatus.getPath)) {
+          PathUsage(dir, true)
+        } else {
+          PathUsage(dir, false)
+        }
+      }
+      .groupBy { case PathUsage(path, _) => path }
+      .map {
+        case (path, usageIter) =>
+          usageIter.reduce { (pu1, pu2) => PathUsage(path, pu1.committed || pu2.committed) }
+      }
+
+    usedDirs.nonEmpty && usedDirs.forall(pu => pu.committed)
```

Comment:
isn't that the opposite of what he has tho? This says the list must be non-empty and all true. I think just forall would let through the empty list